packages/manager/apps/pci-object-storage/src/api/hooks/useStorages.ts

```diff
@@ -0,0 +1,548 @@
+import { useTranslation } from 'react-i18next';
+import { useMemo } from 'react';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import { ColumnSort, PaginationState } from '@ovh-ux/manager-react-components';
+import { ApiError, applyFilters, Filter } from '@ovh-ux/manager-core-api';
+import { useGetProjectRegions } from '@ovh-ux/manager-pci-common';
+import {
+  deleteSwiftContainer,
+  deleteS3Container,
+  getStorages,
+  TStorage,
+  updateStorage,
+  createSwiftStorage,
+  createS3Storage,
+  setContainerAsStatic,
+  setContainerAsPublic,
+  updateStorageType,
+  getStorageAccess,
+  Replication,
+} from '../data/storages';
+import {
+  OBJECT_CONTAINER_MODE_LOCAL_ZONE,
+  OBJECT_CONTAINER_MODE_MONO_ZONE,
+  OBJECT_CONTAINER_MODE_MULTI_ZONES,
+  OBJECT_CONTAINER_OFFER_STORAGE_STANDARD,
+  OBJECT_CONTAINER_OFFER_SWIFT,
+  OBJECT_CONTAINER_TYPE_PUBLIC,
+  OBJECT_CONTAINER_TYPE_STATIC,
+} from '@/constants';
+import { paginateResults } from '@/helpers';
+import { addUser, deleteSwiftObject, TStorageObject } from '../data/objects';
+import { getContainerQueryKey } from './useContainer';
+import { useGetRegion } from './useRegion';
+
+export const sortStorages = (sorting: ColumnSort, storages: TStorage[]) => {
+  const order = sorting.desc ? -1 : 1;
+  switch (sorting?.id) {
+    case 'name':
+    case 'region':
+    case 'mode':
+    case 'offer':
+    case 'containerType':
+      return storages?.sort(
```

Comment:
`sort` mutates the original array, so we need to handle a copy : 
```suggestion
      return storages?.slice().sort(
```
or creating a spread at higher level